test/keycloak-connect-web-spec.js

```diff
@@ -109,49 +117,51 @@ test('Public client should be forbidden for invalid public key', t => {
   return client.then((installation) => {
     installation['realm-public-key'] = TestVector.wrongRealmPublicKey;
     app.build(installation);
-    page.get(app.port);
-
-    return page.output().getText().then(text => {
-      t.equal(text, 'Init Success (Not Authenticated)', 'User should not be authenticated');
-      page.logInButton().click();
-      page.login('test-admin', 'password');
-
-      return page.body().getText().then(text => {
-        t.equal(text, 'Access denied', 'Message should be access denied');
-      }).then(() => {
-        app.destroy();
+    return page.get(app.port).then(() =>
+      page.output().getText().then(text => {
+        t.equal(text, 'Init Success (Not Authenticated)', 'User should not be authenticated');
+        return page.logInButton().click().then(() =>
+          page.login('test-admin', 'password').then(() =>
+            page.body().getText().then(text => {
+              t.equal(text, 'Access denied', 'Message should be access denied');
+            })
+          )
+        );
       })
-        .catch(err => {
-          app.destroy();
-          throw err;
-        });
+    ).then(() => {
+      app.destroy();
+    }).catch(err => {
+      app.destroy();
+      throw err;
     });
   });
 });
 
 test('Confidential client should be forbidden for invalid public key', t => {
-  t.plan(2);
+  t.plan(3);
   var app = new NodeApp();
   var client = admin.createClient(app.confidential('app3'));
 
   return client.then((installation) => {
     installation['realm-public-key'] = TestVector.wrongRealmPublicKey;
     app.build(installation);
-    page.get(app.port);
-
-    return page.output().getText().then(text => {
-      t.equal(text, 'Init Success (Not Authenticated)', 'User should not be authenticated');
-      page.logInButton().click();
-
-      return page.body().getText().then(text => {
-        t.equal(text, 'Access denied', 'Message should be access denied');
-      }).then(() => {
-        app.destroy();
+    return page.get(app.port).then(() =>
+      page.output().getText().then(text => {
+        t.equal(text, 'Init Success (Not Authenticated)', 'User should not be authenticated');
+        return page.logInButton().click().then(() =>
+          page.body().getText().then(text => {
+            t.equal(text, 'Access denied', 'Message should be access denied');
+          })
+            .then(() => page.logout(app.port))
+            .then(() => page.get(app.port, '/check-sso'))
+            .then(() => page.output().getText().then(text => t.equal(text, 'Check SSO Success (Not Authenticated)', 'User should not be authenticated')))
```

Comment:
I moved logout from next test here https://github.com/keycloak/keycloak-nodejs-connect/pull/197/files#diff-97c883a49274cc861db00bd1b5aa7633L163. I introduced another t.equal in order to be sure that the test does not end before logout is finished (since it is called asynchronously).